zerver/views/streams.py

```diff
@@ -159,7 +159,8 @@ def update_stream_backend(
 ) -> HttpResponse:
     # We allow realm administrators to to update the stream name and
     # description even for private streams.
-    stream = access_stream_for_delete_or_update(user_profile, stream_id)
+    stream = get_stream_by_id(stream_id)
+    access_stream_for_admin_actions(user_profile, stream)
```

Comment:
I think everything currently using  `access_stream_for_delete_or_update`  should work with the; so I'm not sure we need a new function rather than just extending the existing function to support being passed non-realm-admins.  I think `access_stream_for_delete_or_update` is probably the better interface (i.e. returning the Stream object)

Reply:
I also initially thought of returning a stream but had not done so as we had to also use this for removing subscriptions and it already had streams. 
But as you suggested for adding check in list_to_streams for removal part, I will change it to return a stream. 

Just wanted to confirm we also wanted to allow stream admin to delete the stream, then we can update access_stream_delete_or_update instead of using different function.

Reply:
I have removed the added function and instead updated the `access_stream_for_delete_or_update` but I am not able to find a way to change it to return a stream object as we use this for checking condition while removing subscribers also and it uses list_to_streams function which uses bulk_get_streams to get streams from stream dict.